ensure_tiller_installed.go

```diff
@@ -395,6 +395,10 @@ func (c *Client) EnsureTillerInstalledWithValues(ctx context.Context, values []s
 			return microerror.Mask(err)
 		}
 	} else if !installTiller && upgradeTiller {
+		if !upgrading {
+			c.logger.LogCtx(ctx, "level", "debug", "message", "found an out-dated tiller but opt not to upgrade it.")
```

Comment:
We can return earlier than this actually, but I just want to place much simpler logic in here.